proxy/http2/Http2ConnectionState.cc

```diff
@@ -1545,10 +1739,14 @@ Http2ConnectionState::release_stream()
         // Can't do this because we just destroyed right here ^,
         // or we can use a local variable to do it.
         // session = nullptr;
+      } else if (shutdown_state == HTTP2_SHUTDOWN_IN_PROGRESS && fini_event == nullptr) {
```

Comment:
This is concerning because it reverts a fix.
See https://github.com/apache/trafficserver/pull/7267

Reply:
Thanks. Good catch. The merge is pretty large and I suppose I accidentally pulled this in thinking that I wanted the additional code, not the removal of it.